client/lib/query-manager/post/index.js

```diff
@@ -0,0 +1,155 @@
+		return every( queryWithDefaults, ( value, key ) => {
+			switch ( key ) {
+				case 'search': {
+					const search = new RegExp( value, 'i' );
+					return search.test( post.title ) || search.test( post.content );
+				}
+
+				case 'after':
+				case 'before':
+				case 'modified_after':
+				case 'modified_before': {
+					const queryDate = moment( value, moment.ISO_8601 );
+					const comparison = /after$/.test( key ) ? 'isAfter' : 'isBefore';
+					const field = /^modified_/.test( key ) ? 'modified' : 'date';
+					return queryDate.isValid() && moment( post[ field ] )[ comparison ]( queryDate );
+				}
+
+				case 'tag':
+				case 'category': {
+					const search = new RegExp( `^${ value }$`, 'i' );
+					const field = 'tag' === key ? 'tags' : 'categories';
+					return some( post[ field ], ( { name, slug } ) => {
+						return search.test( name ) || search.test( slug );
+					} );
+				}
+
+				case 'type':
+					return 'any' === value || value === post.type;
+
+				case 'parent_id':
+					return value === post.parent;
+
+				case 'exclude':
+					if ( Array.isArray( value ) ) {
+						return ! includes( value, post.ID );
+					}
+
+					return value !== post.ID;
+
+				case 'sticky':
+					if ( 'require' === value ) {
+						return post.sticky;
+					} else if ( 'exclude' === value ) {
+						return ! post.sticky;
+					}
+
+					return true;
+
+				case 'author':
+					return get( post, 'author.ID', post.author ) === value;
+
+				case 'status':
+					return value === post[ key ];
+			}
+
+			return true;
+		} );
+	}
+
+	/**
+	 * A sort comparison function that defines the sort order of posts under
+	 * consideration of the specified query.
+	 *
+	 * @param  {Object} query Query object
+	 * @param  {Object} postA First post
+	 * @param  {Object} postB Second post
+	 * @return {Number}       0 if equal, less than 0 if postA is first,
+	 *                        greater than 0 if postB is first.
+	 */
+	sort( query, postA, postB ) {
+		let order;
+
+		switch ( query.order_by ) {
+			case 'ID':
+				order = postA.ID - postB.ID;
+				break;
+
+			case 'comment_count':
+				order = postA.discussion.comment_count - postB.discussion.comment_count;
+				break;
+
+			case 'title':
+				order = postA.title.localeCompare( postB.title );
+				break;
+
+			case 'modified':
+				order = moment( postA.modified ).diff( postB.modified );
+				break;
+
+			case 'date':
+			default:
+				order = moment( postA.date ).diff( postB.date );
+		}
+
+		// Default to descending order, opposite sign of ordered result
+		if ( ! query.order || /^desc$/i.test( query.order ) ) {
+			order *= -1;
+		}
+
+		return order || 0;
+	}
+
+	/**
+	 * Returns a new post after consideration of incoming revision of that
+	 * post. The post can be undefined in the case that the revision is
+	 * new. Optionally patch the post to merge, not replace. Returning
+	 * undefined indicates that post should be removed from known set. Posts
```

Comment:
It's a bit of unfortunate legacy. Not sure the specific reasoning, but this is the behavior of core's `wp_delete_post`:

>wp_delete_post() automatically reverts to `wp_trash_post()` if $force_delete is false, the post_type of `$postid` is page or post, $postid is not already in the trash **and** if that trash feature enabled (which it it is by default).

https://codex.wordpress.org/Function_Reference/wp_delete_post